core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,332 @@
+                    ' received %s' % type(self.assigned_reviewer_id))
+
+        if not isinstance(self.final_reviewer_id, basestring):
+            if self.final_reviewer_id:
+                raise utils.ValidationError(
+                    'Expected final_reviewer_id to be a string, received %s' %
+                    type(self.final_reviewer_id))
+
+        if not isinstance(self.change_cmd, dict):
+            raise utils.ValidationError(
+                'Expected change_cmd to be a dict, received %s' % type(
+                    self.change_cmd))
+
+        if not isinstance(self.score_category, basestring):
+            raise utils.ValidationError(
+                'Expected score_category to be a string, received %s' % type(
+                    self.score_category))
+
+        if (
+                suggestion_models.SCORE_CATEGORY_DELIMITER not in
+                self.score_category):
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if (
+                len(self.score_category.split(
+                    suggestion_models.SCORE_CATEGORY_DELIMITER))) != 2:
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if self.get_score_type() not in suggestion_models.SCORE_TYPE_CHOICES:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be among allowed'
+                ' choices, received %s' % self.get_score_type())
+
+    def accept(self):
+        """Accepts the suggestion. Each subclass must implement this function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement accept.')
+
+    def update_change_cmd_before_accept(self):
+        """Before accepting the suggestion, some parts of the change_cmd might
+        need to be updated. Each subclass must implement this function as
+        applicable.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement '
+            'update_change_cmd_before_accept.')
+
+
+class SuggestionEditStateContent(BaseSuggestion):
+    """Domain object for a suggestion of type
+    SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+    """
+
+    def __init__( # pylint: disable=super-init-not-called
+            self, suggestion_id, target_id, target_version_at_submission,
+            status, author_id, assigned_reviewer_id, final_reviewer_id,
+            change_cmd, score_category):
+        """Initializes a Suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
```

Comment:
The object is of type SuggestionEditStateContent. Perhaps, it is better to say "Initializes an object of type SuggestionEditStateContent corresponding to the SUGGESTION_TYPE_EDIT_STATE_CONTENT choice" or something along those lines.

Reply:
Done